utils/core/src/main/kotlin/OkHttpClientHelper.kt

```diff
@@ -140,43 +141,47 @@ object OkHttpClientHelper {
     suspend fun await(request: Request, block: BuilderConfiguration? = null): Response =
         buildClient(block).newCall(request).await()
 
+    /**
+     * Download from [url] with optional [acceptEncoding] and return a [Result] with the [Response] and non-nullable
+     * [ResponseBody] on success, or a [Result] wrapping an [IOException] (which might be a [HttpDownloadError]) on
+     * failure.
+     */
+    fun download(url: String, acceptEncoding: String? = null): Result<Pair<Response, ResponseBody>> {
+        val request = Request.Builder()
+            .apply { acceptEncoding?.also { header("Accept-Encoding", it) } }
+            .get()
+            .url(url)
+            .build()
+
+        return runCatching { execute(request) }.mapCatching { response ->
+            val body = response.body
+
+            if (!response.isSuccessful || body == null) {
```

Comment:
This seems to alter the behavior of the `downloadText()` function, previously it returned `Success` if the body was `null`.

Reply:
Well spotted :-) This actually is an alignment with the behavior of `downloadFile()`, and in `downloadText()` it was "wrong" from start, as I lazily just used `.orEmpty()`. I'll document this in the commit message.

Reply:
Yes, that's why I asked, because you did the other scanners in the previous commit. But I don't care too much so I'll approve.

Reply:
Ok, I just shortly checked the callers of `downloadText()` and it seems none of them could work with an empty string anyway.